core/trino-main/src/test/java/io/trino/sql/query/TestFilterInaccessibleColumns.java

```diff
@@ -0,0 +1,197 @@
+        assertThat(assertions.query("SELECT * FROM nation WHERE name = 'FRANCE'"))
+                .matches("VALUES (BIGINT '-6',CAST('FRANCE' AS VARCHAR(25)), BIGINT '3', CAST('refully final requests. regular, ironi' AS VARCHAR(152)))");
+    }
+
+    /**
+     * Test filtering when columns are explicitly specified in SELECT
+     */
+    @Test
+    public void testFilterExplicitSelect()
+    {
+        // Select the columns that are available to us explicitly
+        accessControl.reset();
+        accessControl.deny(privilege("nation.comment", SELECT_COLUMN));
+        assertThat(assertions.query("SELECT nationkey, name, regionkey FROM nation WHERE name = 'FRANCE'"))
+                .matches("VALUES (BIGINT '6', CAST('FRANCE' AS VARCHAR(25)), BIGINT '3')");
+
+        // Select all columns explicitly
+        accessControl.reset();
+        accessControl.deny(privilege("nation.comment", SELECT_COLUMN));
+        assertThatThrownBy(() -> assertions.query("SELECT nationkey, name, regionkey, comment FROM nation WHERE name = 'FRANCE'"))
+                .hasMessage("Access Denied: Cannot select from columns [nationkey, regionkey, name, comment] in table or view local.tiny.nation");
+    }
+
+    @Test
+    public void testPredicateOnInaccessibleColumn()
+    {
+        // Hide name but use it in the query predicate
+        accessControl.reset();
+        accessControl.deny(privilege("nation.name", SELECT_COLUMN));
+        assertThatThrownBy(() -> assertions.query("SELECT * FROM nation WHERE name = 'FRANCE'"))
+                .hasMessage("Access Denied: Cannot select from columns [nationkey, regionkey, name, comment] in table or view local.tiny.nation");
+    }
+
+    /**
+     * Test that filtering on INFORMATION_SCHEMA tables works as expected.
+     */
+    @Test
+    public void testFilterInformationSchema()
+    {
+        accessControl.reset();
+        accessControl.deny(privilege("columns.table_catalog", SELECT_COLUMN));
+        accessControl.deny(privilege("columns.ordinal_position", SELECT_COLUMN));
+        accessControl.deny(privilege("columns.column_default", SELECT_COLUMN));
+        accessControl.deny(privilege("columns.is_nullable", SELECT_COLUMN));
+        accessControl.deny(privilege("columns.data_type", SELECT_COLUMN));
+        // SHOW COLUMNS and DESCRIBE fail since data_type isn't visible
+        assertThatThrownBy(() -> assertions.query("SHOW COLUMNS FROM nation")).hasMessage("Access Denied: Cannot select from columns [ordinal_position, extra_info, table_schema, column_name, data_type, comment, table_name] in table or view local.information_schema.columns");
+        assertThatThrownBy(() -> assertions.query("DESCRIBE nation")).hasMessage("Access Denied: Cannot select from columns [ordinal_position, extra_info, table_schema, column_name, data_type, comment, table_name] in table or view local.information_schema.columns");
```

Comment:
This query should not fail. If you are able to query table then surely you should be able to to `DESCRIBE` it. Columns that got hidden should be simply filtered here. Same comment for `SHOW COLUMNS FROM nation`